sdk-java/src/main/java/ly/count/sdk/java/internal/ModuleViews.java

```diff
@@ -1,9 +1,497 @@
+        }
+
+        return viewID;
+    }
+
+    void addSegmentationToViewWithNameInternal(@Nullable String viewName, @Nullable Map<String, Object> viewSegmentation) {
+        String viewID = validateViewWithName(viewName, "addSegmentationToViewWithNameInternal");
+        if (viewID == null) {
+            return;
+        }
+        addSegmentationToViewWithIDInternal(viewID, viewSegmentation);
+    }
+
+    private void addSegmentationToViewWithIDInternal(String viewID, Map<String, Object> viewSegmentation) {
+        ViewData vd = validateViewID(viewID, "addSegmentationToViewWithIdInternal");
+        if (vd == null) {
+            return;
+        }
+
+        if (viewSegmentation == null || viewSegmentation.isEmpty()) {
+            L.e("[ModuleViews] addSegmentationToViewWithIdInternal, Trying to record view with null or empty view segmentation, ignoring request");
+            return;
+        }
+        removeReservedKeysFromViewSegmentation(viewSegmentation);
+        vd.viewSegmentation.putAll(viewSegmentation);
+    }
+
+    /**
+     * Creates a crypto-safe SHA-256 hashed random value
+     *
+     * @return returns a random string value
+     */
+    public static String safeRandomVal() {
+        long timestamp = System.currentTimeMillis();
+        SecureRandom random = new SecureRandom();
+        byte[] value = new byte[6];
+        random.nextBytes(value);
+        String b64Value = Utils.Base64.encode(value);
+        return b64Value + timestamp;
+    }
+
+    public class Views {
+
+        /**
+         * Record a view manually, without automatic tracking
+         * or tracks a view that is not automatically tracked
+         * like a fragment, Message box or a transparent Activity
+         *
+         * @param viewName String - name of the view
+         * @return Returns View ID
+         */
+        public String startAutoStoppedView(@Nullable String viewName) {
+            synchronized (Countly.instance()) {
+                // call the general function that has two parameters
+                return startAutoStoppedView(viewName, null);
+            }
+        }
+
+        /**
+         * Record a view manually, without automatic tracking
+         * or tracks a view that is not automatically tracked
+         * like a fragment, Message box or a transparent Activity
+         * with segmentation. (This is the main function that is used)
+         *
+         * @param viewName String - name of the view
+         * @param viewSegmentation Map<String, Object> - segmentation that will be added to the view, set 'null' if none should be added
+         * @return String - view ID
+         */
+        public String startAutoStoppedView(@Nullable String viewName, @Nullable Map<String, Object> viewSegmentation) {
+            synchronized (Countly.instance()) {
+                L.i("[Views] Calling startAutoStoppedView [" + viewName + "]");
```

Comment:
I know this is the android code but feels like we are missing out logs about segmentation. Sth similar to startView might be better. my 2 cents for you guys